traffic_ops/traffic_ops_golang/api/shared_handlers.go

```diff
@@ -395,34 +397,125 @@ func CreateHandler(creator Creator) http.HandlerFunc {
 		obj := reflect.New(objectType).Interface().(Creator)
 		obj.SetInfo(inf)
 
-		err := decodeAndValidateRequestBody(r, obj)
-		if err != nil {
-			HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, err, nil)
-			return
-		}
+		if c, ok := obj.(MultipleCreator); ok && c.AllowMultipleCreates() {
+			data, err := ioutil.ReadAll(r.Body)
+			if err != nil {
+				HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, err, nil)
+				return
+			}
 
-		if t, ok := obj.(Tenantable); ok {
-			authorized, err := t.IsTenantAuthorized(inf.User)
+			objSlice, err := parseMultipleCreates(data, objectType, inf)
 			if err != nil {
-				HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("checking tenant authorized: "+err.Error()))
+				HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
 				return
 			}
-			if !authorized {
-				HandleErr(w, r, inf.Tx.Tx, http.StatusForbidden, errors.New("not authorized on this tenant"), nil)
+
+			for _, objElemInt := range objSlice {
```

Comment:
this looks great! but I wonder if there is a different approach so we dont have to loop through and do n db calls but just one to do a bulk insert. Not blocking but more of a comment to think about.